plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -113,14 +115,25 @@ uncheckedDeserialiseUPLC = unSerialiseViaFlat . deserialise . BSL.fromStrict . f
   where
     unSerialiseViaFlat (SerialiseViaFlat (UPLC.UnrestrictedProgram a)) = a
 
--- | A variant of `Script` with a specialized decoder.
-newtype ScriptForExecution = ScriptForExecution (UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ())
+-- | A script with named de-bruijn indices.
+newtype ScriptNamedDeBruijn = ScriptNamedDeBruijn (UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ())
+
+-- | A Plutus script ready to be evaluated on-chain, via @evaluateScriptRestricting@.
+data ScriptForEvaluation = UnsafeScriptForEvaluation SerialisedScript ScriptNamedDeBruijn
```

Comment:
We need a few instances:

`NFData` instance is required for benchmarking.

Also `NoThunks` and `Eq` will be needed.

`Show` would be nice, but not an absolute requirement.